### PR TITLE
ensure AKS immutable configuration has webhook enforcement

### PR DESCRIPTION
**Pull request by jackfrancis**
**What type of PR is this?**

/kind bug
/kind cleanup

**What this PR does / why we need it**:

This PR replaces a few, one-off validation implementations to re-use common funcs instead.

More importantly, the following `AzureManagedControlPlane` properties have been fixed for strict immutability as according the requirements of the AKS API for those properties:

- `Spec.SSHPublicKey`
- `Spec.DNSServiceIP`
- `Spec.NetworkPlugin`
- `Spec.NetworkPolicy`
- `Spec.LoadBalancerSKU`

And the following `AzureManagedMachinePool` properties as well:

- `Spec.OSDiskSizeGB`
- `Spec.MaxPods`
- `Spec.OsDiskType`

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
ensure AKS immutable configuration has webhook enforcement
```

### CODE REVIEW

File: util/webhook/validator.go [hunk 1]
```diff
@@ -167,3 +168,22 @@ func ValidateImmutable(path *field.Path, oldVal, newVal any) *field.Error {
 
 	return nil
 }
+
+// EnsureStringSlicesAreEquivalent returns if two string slices have equal lengths,
+// and that they have the exact same items; it does not enforce strict ordering of items.
+func EnsureStringSlicesAreEquivalent(a []string, b []string) bool {
```
Comment (jackfrancis) [inline util/webhook/validator.go:174]: This new func is in response to a comment from @nojnhuh [here](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2718#discussion_r992699827) that the existing `ensureStringSlicesAreEqual` func has a bug (and isn't unit tested).

**Review by mboersma — COMMENTED** [review]
/lgtm

Nice refactoring! This is simpler and more readable by far.

**Comment by jackfrancis** [comment]
I labeled this `kind/bug` because strictly speaking the subtle changes to immutability across the described properties reflect a bug in the way we're dealing with the AKS API, but I would advocate *we not* cherry-pick these changes.

Unless we hear from folks who are running into these edge cases in the wild.

Thoughts @mboersma @CecileRobertMichon @nojnhuh ?

**Comment by mboersma** [comment]
> ...I would advocate we not cherry-pick these changes.
>
> Unless we hear from folks who are running into these edge cases in the wild.

Makes sense to me.

**Pull request merged**